CMakeLists.txt

```diff
@@ -19,9 +19,13 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-option(NOTEST "Disable all tests" OFF)
+option(BUILD_TESTS "Build tests" ON)
```

Comment:
Do you think, rather than having our own flag, we can use the built-in method for enabling the build of tests?  https://cliutils.gitlab.io/modern-cmake/chapters/testing.html